libcodechecker/libclient/thrift_call.py

```diff
@@ -0,0 +1,86 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+import os
+import socket
+import sys
+
+from thrift.protocol.TProtocol import TProtocolException
+from thrift.Thrift import TApplicationException
+
+import shared
+from libcodechecker.logger import get_logger
+
+LOG = get_logger('system')
+
+
+def ThriftClientCall(function):
+    """ Wrapper function for thrift client calls.
+        - open and close transport,
+        - log and handle errors
+    """
+    # LOG.debug(type(function))
+    funcName = function.__name__
+
+    def wrapper(self, *args, **kwargs):
+        # LOG.debug('['+self.__host+':'+str(self.__port)+'] '
+        #       '>>>>> ['+funcName+']')
+        # before = datetime.datetime.now()
+        self.transport.open()
+        func = getattr(self.client, funcName)
+        try:
+            res = func(*args, **kwargs)
+            return res
+        except shared.ttypes.RequestFailed as reqfailure:
```

Comment:
Do not we want to handle `API_MISMATCH` here too?
https://github.com/Ericsson/codechecker/blob/98f4ccd541566f43dd358d84321dc3056436b79c/libcodechecker/libclient/authentication_helper.py#L68

Reply:
done